cmd/kubeadm/app/phases/certs/renewal/manager.go

```diff
@@ -97,6 +101,15 @@ func NewManager(cfg *kubeadmapi.ClusterConfiguration, kubernetesDir string) (*Ma
 				readwriter: pkiReadWriter,
 			}
 		}
+
+		pkiReadWriter := newPKICertificateReadWriter(rm.cfg.CertificatesDir, ca.BaseName)
+		rm.certificateAuthorities[ca.Name] = &CertificateRenewHandler{
```

Comment:
For CA we are not handling the renewal, so using a `CertificateRenewHandler` might be confusing.
Also, `CertificateRenewHandler` was designed for Certificates, and things like `IsExternallyManaged` should be managed in a slightly different way

Reply:
i agree that renewal should not apply for CA.